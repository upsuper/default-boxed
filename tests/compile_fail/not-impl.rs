use default_boxed::DefaultBoxed;

struct Bar;

#[derive(DefaultBoxed)]
struct Foo {
    a: Bar,
}

fn main() {}
