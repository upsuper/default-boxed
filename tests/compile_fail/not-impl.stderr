error[E0277]: the trait bound `Bar: std::default::Default` is not satisfied
 --> $DIR/not-impl.rs:7:8
  |
7 |     a: Bar,
  |        ^^^ the trait `std::default::Default` is not implemented for `Bar`
  |
  = note: required because of the requirements on the impl of `default_boxed::DefaultBoxed` for `Bar`
  = note: required by `default_boxed::DefaultBoxed::default_in_place`
